src/backends/DomRenderer.js

```diff
@@ -36,12 +36,35 @@ function setStyle(rule: HTMLElement|CSSStyleRule, prop: string, value: string):
   return true
 }
 
+function extractSelector(cssText: string, from: number = 0) {
+  return cssText.substr(from, cssText.indexOf('{') - 1)
+}
+
+const CSSRuleTypes = {
+  STYLE_RULE: 1,
+  KEYFRAMES_RULE: 7
+}
 
 /**
  * Get the selector.
  */
-function getSelector(rule: CSSStyleRule): string {
-  return rule.selectorText
+function getSelector(rule: CSSOMRule): string {
+  if (rule.type === CSSRuleTypes.STYLE_RULE) return rule.selectorText
```

Comment:
have you checked if rule.type is a getter? If it is - lets not call it more than once.